tests/optional/programs/optional/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+//! This example demonstrates the ability to use optional accounts in
+//! structs deriving `Accounts`.
+
+use anchor_lang::prelude::*;
+pub use context::*;
+
+pub mod account;
+pub mod context;
+declare_id!("FNqz6pqLAwvMSds2FYjR4nKV3moVpPNtvkfGFrqLKrgG");
+
+#[program]
+mod optional {
+    use super::*;
+
+    pub fn initialize(ctx: Context<Initialize>, value: u64, key: Pubkey) -> Result<()> {
+        let optional_pda = &mut ctx.accounts.optional_pda;
```

Comment:
caveat, i'm bad at rust

- this is a mutable reference only to the optional_pda which is `Option<Account<'info, DataAccount>>`, it's not a mutable reference to `Account<'info, DataAccount>` right? This is seems like an awkward interface to use for accessing the underlying accounts?